### ISSUE TITLE
Validation.orElse equivalent for transforming Successes.

### ISSUE THREAD

**Issue by Undistraction** [origamitower/folktale#175]
Folktale currently offers `orElse` to use when transforming the value of a validation that is a Failure. However there is no analogue for transforming the value of a Success. 

If I want to transform a validation if it is a failure I can do:
```
validation.orElse(...);
```

However, to transform a validation if it is a success I have to do:
```
validation.matchWith({
  Success: value => ...
  Failure: identity,
})
```

This feels a little uneven to me. How about adding a method that is the equivalent of `orElse`, but is only executed if the validation is a Success?

**Comment by robotlolita** [comment]
A way of sequencing successful validations will be added in the next version, though that might take a couple of months

**Comment by Undistraction** [comment]
@robotlolita Great. Thanks for all your work.

**Comment by robotlolita** [comment]
This has been released as `folktale@2.3.0`.
